test/js/substr.spec.js

```diff
@@ -0,0 +1,242 @@
+/**
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
+* This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+**/
+
+const { TaintedUtils, taintFormattedString, formatTaintedValue } = require('./util')
+const assert = require('assert')
+
+const rangesTestCases = [
+  {
+    source: ':+-foobarbaz-+:',
+    result: '',
+    start: 3,
+    end: 0
+  },
+  {
+    source: ':+-foobarbaz-+:',
+    result: '',
+    start: -6,
+    end: -3
+  },
+  {
+    source: ':+-foobarbaz-+:',
+    result: ':+-foo-+:',
+    start: 0,
+    end: 3
+  },
+  {
+    source: ':+-foobarbaz-+:',
+    result: ':+-bar-+:',
+    start: -6,
+    end: 3
+  },
+  {
+    source: ':+-foobarbaz-+:',
+    result: ':+-foo-+:',
+    start: -9,
+    end: 3
+  },
+  {
+    source: ':+-foobarbaz-+:',
+    result: ':+-barbaz-+:',
+    start: 3,
+    end: 6
+  },
+  {
+    source: ':+-foobarbaz-+:',
+    result: ':+-baz-+:',
+    start: 6,
+    end: 9
+  },
+  {
+    source: ':+-foobarbaz-+:',
+    result: ':+-baz-+:',
+    start: 6
+  },
+  {
+    source: 'foo:+-bar-+:baz',
+    result: 'foo',
+    start: 0,
+    end: 3
+  },
+  {
+    source: 'foo:+-bar-+:baz',
+    result: 'oo:+-bar-+:',
+    start: 1,
+    end: 5
+  },
+  {
+    source: 'foo:+-bar-+:baz',
+    result: ':+-r-+:baz',
+    start: 5,
+    end: 9
+  },
+  {
+    source: 'foo:+-bar-+:baz',
+    result: 'baz',
+    start: 6,
+    end: 9
+  },
+  {
+    source: 'foo:+-bar-+:b:+-az-+:',
+    result: 'b:+-az-+:',
+    start: 6,
+    end: 9
+  },
+  {
+    source: 'foo:+-bar-+:b:+-a-+:z',
+    result: 'b:+-a-+:z',
+    start: 6,
+    end: 9
+  },
+  {
+    source: 'foo:+-bar-+:b:+-a-+:z',
+    result: 'b:+-a-+:z',
+    start: 6,
+    end: 15
+  },
+  {
+    source: 'foo:+-bar-+:b:+-a-+:z:+-z-+::+-z-+:',
+    result: 'b:+-a-+:z:+-z-+::+-z-+:',
+    start: 6,
+    end: 9
+  },
+  {
+    source: 'foo:+-bar-+:b:+-a-+:z:+-z-+::+-z-+:',
+    result: 'b:+-a-+:z:+-z-+::+-z-+:',
+    start: 6,
+    end: 10
+  },
+  {
+    source: 'foo:+-bar-+:b:+-a-+:z:+-z-+::+-z-+:',
+    result: 'b:+-a-+:z:+-z-+::+-z-+:',
+    start: 6,
+    end: 20
+  }
+]
+
+function substringCheckRanges (id, testString, start, end, expected) {
```

Comment:
We are in substr test
```suggestion
function substrCheckRanges (id, testString, start, end, expected) {
```